lib/jsender/json.rb

```diff
@@ -11,12 +11,12 @@ def error(message: "An error has occurred")
       }.to_json
     end
 
-    def failure(message: "A failure has occurred")
+    def failure(message: "A failure has occurred", data: {})
       {
         'status' => 'fail',
         'data' => {
          'message' => message
-        }
+        }.merge(data)
```

Comment:
I'd suggest using supplied data if given, otherwise the generic validation failure message. Mixing data and a message will probably trip up front-end consumers when they expect data to be a dictionary of params that failed validation.